File: typescript-examples/03-agent-streaming/agents.ts
/**
 * Agents for the streaming example.
 */

import { defineAgent, maxSteps } from '@polos/sdk';
import { anthropic } from '@ai-sdk/anthropic';

// Storyteller agent - good for demonstrating streaming with longer outputs
export const storyteller = defineAgent({
  id: 'storyteller',
  model: anthropic('claude-sonnet-4-5'),
  systemPrompt:
    `You are a creative storyteller. When asked for a story, ` +
    `tell an engaging, vivid story with descriptions and dialogue. ` +
    `Keep stories between 200-400 words unless asked for a different length.`,
  stopConditions: [maxSteps({ count: 5 })],
});

File: python-examples/03-agent-streaming/agents.py
"""Agents for the streaming example."""

from polos import Agent, max_steps, MaxStepsConfig


# Storyteller agent - good for demonstrating streaming with longer outputs
storyteller = Agent(
    id="storyteller",
    provider="anthropic",
    model="claude-sonnet-4-5",
    system_prompt="""You are a creative storyteller. When asked for a story,
tell an engaging, vivid story with descriptions and dialogue.
Keep stories between 200-400 words unless asked for a different length.""",
    stop_conditions=[
        max_steps(MaxStepsConfig(count=5)),
    ],
)
